src/main/kotlin/cacophony/automata/DFA.kt

```diff
@@ -20,3 +20,30 @@ interface DFA<DFAState> {
     // Returned value is map accepting current state and symbol, and returning new state, which may not exist.
     fun getProductions(): Map<Pair<DFAState, Char>, DFAState>
 }
+
+public data class SimpleDFA<StateType>(
```

Comment:
If you created it maybe you add at least one test to see if at least getAllStates is correct, it's like the only place with some logic.